### PR TITLE
Calendar API To returns checkmarks for provided user & returns whole …

### PR DESCRIPTION
**Pull request by miguelcard**
…members objects in spaceReadWithHabbit Serializer

### CODE REVIEW

File: accountability_project/Models/spaces/api/serializers.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ class SpaceSerializerToReadWithHabits(serializers.ModelSerializer):
     members_count = serializers.IntegerField(read_only=True)
     habits_count = serializers.IntegerField(read_only=True)
     # shows the detailed habits that belong to this space 
+    members = serializers.SerializerMethodField(method_name='get_members')
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/serializers.py:71]: - [ ] Please change the name of this class to SpaceSerializerToReadWithHabitsAndMembers -> to be more explicit
- [ ] also, something like this does not work in this case? `members = SimpleUserSerializer(many=True, read_only=True)`

File: accountability_project/Models/spaces/api/serializers.py [hunk 2]
```diff
@@ -68,6 +68,7 @@ class SpaceSerializerToReadWithHabits(serializers.ModelSerializer):
     members_count = serializers.IntegerField(read_only=True)
     habits_count = serializers.IntegerField(read_only=True)
     # shows the detailed habits that belong to this space
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/serializers.py:70]: - [ ] this comment should go one line below/under your new code (its there to describe the space_habits attribute)

File: accountability_project/Models/spaces/api/api.py [hunk 3]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
         logger.info(f' Get all users from space with id: {space_id}')
         space = get_object_or_404(Space, id=space_id, members=self.request.user)
         logger.info(f' Retrieving all users for space with id: {space_id} and name: {space.name}')
-        return space.members.all()
\ No newline at end of file
+        return space.members.all()
+
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:224]: just to keep it consistent as the views above :) 
```suggestion

# GET
```

File: accountability_project/Models/spaces/api/api.py [hunk 4]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
...
+class CalendarAPIView(generics.ListAPIView):
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:225]: - [ ] Please add the description/documentation of this view like the views above, this proves helpful for the automatically generated API documentation

File: accountability_project/Models/spaces/api/api.py [hunk 5]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
...
+    serializer_class = RecurrentHabitSerializerToRead
+
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:227]: - [ ] please add pagination and take care of the frontend accordingly (not sure if this suggestion has the correct indentation, watch out)

```suggestion
    pagination_class = SpaceHabitsPagination

```

File: accountability_project/Models/spaces/api/api.py [hunk 6]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
...
+    def get_queryset(self):
+        space_id = self.kwargs.get('pk')
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:229]: - [ ] please add this logging below this line for better traceability:

`logger.info(f' Get all recurrent habits from space with id: {space_id}')`

File: accountability_project/Models/spaces/api/api.py [hunk 7]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
...
+        space = get_object_or_404(Space, id=space_id)
+        owner_id = self.kwargs.get('owner_pk')
+        owner = get_object_or_404(User, id=owner_id)
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:232]: - [ ] please add this logging below this line for better traceability:

`logger.info(f' Retrieving all recurrent habits for user with id: {owner_id} in space: {space_id} {space.name}')`

File: accountability_project/Models/spaces/api/api.py [hunk 8]
```diff
@@ -219,4 +220,14 @@ def get_queryset(self):
...
-        return space.members.all()
\ No newline at end of file
+        return space.members.all()
+
+class CalendarAPIView(generics.ListAPIView):
+    serializer_class = RecurrentHabitSerializerToRead
+
+    def get_queryset(self):
+        space_id = self.kwargs.get('pk')
+        space = get_object_or_404(Space, id=space_id)
```
Comment (miguelcard) [inline accountability_project/Models/spaces/api/api.py:230]: - [ ] here also filter the space with:

` members=self.request.user`
this means only users who belong (are members) to this space will be able to query this information, we dont want any user to see the habits of another user if they dont belong to the same space.

**Comment by mutant46** [comment]
All points covered, will push the changes after testing with frontend.

**Pull request merged**